### PR TITLE
Add options to scanners (fix #311)

### LINKED ISSUE #311: Normalise "options" usage in scanners
This will help with #190 and keeps our method of supplying scanner classes with arbitrary, extra info (eg things other than contents or filename) consistent.

### PR DESCRIPTION
**Pull request by tofumatt**
Started this while doing #190, realised we needed a consistent way to pass in extra data to scanners and this seemed the best way; we can pass all options right onto the rules with this.

### CODE REVIEW

File: src/scanners/css.js [hunk 1]
```diff
@@ -31,25 +31,23 @@ export default class CSSScanner extends BaseScanner {
     }
 
     var file = cssCode.position.source;
-    var startLine = cssCode.position.start.line;
-    var startColumn = cssCode.position.start.column;
+    var options = Object.assign(this.options, {
```
Comment (muffinresearch) [inline src/scanners/css.js]: This is mutating an instance level options object each time - I'm not too sure about that.

Comment (muffinresearch) [inline src/scanners/css.js]: It might be better to call this something else and the use `Object.assign({}, this.options, { blah })`

Comment (tofumatt) [inline src/scanners/css.js]: Oh, you're right, my bad. That was a total oversight. Thanks! :+1:

File: src/scanners/css.js [hunk 2]
```diff
@@ -31,25 +31,23 @@ export default class CSSScanner extends BaseScanner {
     }
 
     var file = cssCode.position.source;
-    var startLine = cssCode.position.start.line;
-    var startColumn = cssCode.position.start.column;
+    var cssInfo = Object.assign({}, this.options, {
```
Comment (muffinresearch) [inline src/scanners/css.js]: Maybe `opts` or `cssOptions` would be better as this is called options in the actual rules?

**Comment by muffinresearch** [comment]
r+wc

File: src/scanners/css.js (hunk 2 above)
Comment (tofumatt) [inline src/scanners/css.js]: Sure thing; I’ll fix and merge.

**Pull request merged**